### PR TITLE
fix: mount Docker credentials

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

It adds the Docker credentials to the Docker compose.

## Why is it important?

We need to login in our registry to grab some Docker images. Removes the `docker pull` introduced on https://github.com/elastic/beats/pull/17620

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -660,13 +660,13 @@ def withBeatsEnv(boolean archive, Closure body) {
...
     unstash 'source'
-    if(os == 'linux'){
+    if(isDockerInstalled()){
       dockerLogin(secret: "${DOCKERELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
       // FIXME workaround until we fix the packer cache
       // Retry to avoid DDoS detection from the server
       retry(3) {
         sleep randomNumber(min: 2, max: 5)
-        sh 'docker pull docker.elastic.co/observability-ci/database-enterprise:12.2.0.1'
+        //sh 'docker pull docker.elastic.co/observability-ci/database-enterprise:12.2.0.1'
```
Comment (cachedout) [inline Jenkinsfile:669]: Is this intentional? This PR seems like it's just about adding the credentials and not about disabling docker pulls.

Comment (kuisathaverat) [inline Jenkinsfile:669]:
```suggestion
```

File: Jenkinsfile [hunk 2]
```diff
@@ -660,13 +660,13 @@ def withBeatsEnv(boolean archive, Closure body) {
   ]) {
     deleteDir()
     unstash 'source'
-    if(os == 'linux'){
+    if(isDockerInstalled()){
       dockerLogin(secret: "${DOCKERELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
       // FIXME workaround until we fix the packer cache
       // Retry to avoid DDoS detection from the server
       retry(3) {
         sleep randomNumber(min: 2, max: 5)
```
Comment (kuisathaverat) [inline Jenkinsfile:668]:
```suggestion
```

File: Jenkinsfile [hunk 3]
```diff
@@ -660,13 +660,13 @@ def withBeatsEnv(boolean archive, Closure body) {
   ]) {
     deleteDir()
     unstash 'source'
-    if(os == 'linux'){
+    if(isDockerInstalled()){
       dockerLogin(secret: "${DOCKERELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
       // FIXME workaround until we fix the packer cache
       // Retry to avoid DDoS detection from the server
       retry(3) {
```
Comment (kuisathaverat) [inline Jenkinsfile:667]:
```suggestion
```

File: Jenkinsfile [hunk 4]
```diff
@@ -660,13 +660,13 @@ def withBeatsEnv(boolean archive, Closure body) {
...
         sleep randomNumber(min: 2, max: 5)
-        sh 'docker pull docker.elastic.co/observability-ci/database-enterprise:12.2.0.1'
+        //sh 'docker pull docker.elastic.co/observability-ci/database-enterprise:12.2.0.1'
       }
```
Comment (kuisathaverat) [inline Jenkinsfile:670]:
```suggestion
```

File: Jenkinsfile [hunk 5]
```diff
@@ -660,14 +660,10 @@ def withBeatsEnv(boolean archive, Closure body) {
   ]) {
     deleteDir()
     unstash 'source'
-    if(os == 'linux'){
+    if(isDockerInstalled()){
       dockerLogin(secret: "${DOCKERELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
       // FIXME workaround until we fix the packer cache
```
Comment (kuisathaverat) [inline Jenkinsfile:665]:
```suggestion
```

File: Jenkinsfile [hunk 6]
```diff
@@ -660,14 +660,10 @@ def withBeatsEnv(boolean archive, Closure body) {
...
       // Retry to avoid DDoS detection from the server
```
Comment (kuisathaverat) [inline Jenkinsfile:666]:
```suggestion
```

File: Jenkinsfile (hunk 1 above)
Comment (kuisathaverat) [reply Jenkinsfile]: this PR removes the need of the workaround, so I've removed those lines

File: x-pack/metricbeat/docker-compose.yml [hunk 7]
```diff
@@ -10,6 +10,7 @@ services:
     volumes:
       - ${PWD}/../..:/go/src/github.com/elastic/beats/
       - /var/run/docker.sock:/var/run/docker.sock
+      - ${HOME}/.docker:/root/.docker:ro
```
Comment (urso) [inline x-pack/metricbeat/docker-compose.yml:13]: Fun fact, `PWD` and `HOME` are not available in docker-compose on windows. If we want to be able to run full system tests in the future on windows as well, we will have to do some modifications.

Comment (kuisathaverat) [reply x-pack/metricbeat/docker-compose.yml:13]: yep both env vars impact developers on local laptops that are testing `x-pack/metricbeat`, so there were impacted by `${PWD}` before this PR, and not they are impacted too by `${HOME}`.
Also, any windows developer running Docker configured for windows containers cannot run the tests (windows only allow to install windows containers support or Linux containers support not both).

Comment (urso) [reply x-pack/metricbeat/docker-compose.yml:13]: > Also, any windows developer running Docker configured for windows containers cannot run the tests (windows only allow to install windows containers support or Linux containers support not both).

Right. But docker-compose is evaluated from the POV of the host. So it doesn't work with linux containers on a windows host (I tried :) ). But this is not new. Replacing `PWD` with `.` helps btw.

The main impact is not `${PWD}` or `${HOME}` per se. But developers will need to create a `.docker` file/directory. What shall we include there in order to test locally?

Comment (kuisathaverat) [reply x-pack/metricbeat/docker-compose.yml:13]: IIRC `/var/run/docker.sock` does not work on windows too. The compose file only works on Linux right now, add the Docker credendial mounted on the containers will work on any case if the file does not exist will mount nothing. The Oracle test is enabled by an env var disabled by default, so I think it is not an issue to add the Docker folder to the container.

**Pull request merged**